nanoFramework.Hardware.Esp32.Rmt/ReceiverChannelSettings.cs

```diff
@@ -0,0 +1,109 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+
+namespace nanoFramework.Hardware.Esp32.Rmt
+{
+    /// <summary>
+    /// <see cref="ReceiverChannel"/>'s settings class.
+    /// </summary>
+    /// <remarks>
+    /// All changes made to properties of this class are ignored after the <see cref="ReceiverChannel"/> is initialized.
+    /// The equivalent properties in the channel instance can be used to make on-the-fly changes to the configurations.
+    /// </remarks>
+    public sealed class ReceiverChannelSettings : RmtChannelSettings
+    {
+        private ushort _idleThreshold;
+        private bool _enableFilter;
+        private byte _filterThreshold;
+        private TimeSpan _receiveTimeout;
+
+        /// <summary>
+        /// Gets or sets the idle threshold after which the receiver will go into idle mode 
+        /// and <see cref="RmtCommand"/>s are copied into the ring buffer and availble to your code.
+        /// </summary>
+        /// <remarks>
+        /// The receive process finishes(goes idle) when no edges have been detected for Idle Threshold clock cycles.
+        /// Supported value range between 1 and 65535 (0xFFFF)
+        /// </remarks>
+        public ushort IdleThreshold
+        {
+            get => _idleThreshold;
+            set
+            {
+                if (value == 0)
+                {
+                    throw new ArgumentOutOfRangeException();
+                }
+
+                _idleThreshold = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the filter state. 
+        /// If enabled, the receiver will ignore pulses with widths less than specified in <see cref="FilterThreshold"/>.
+        /// </summary>
+        public bool EnableFilter 
+        { 
+            get => _enableFilter; 
+            set => _enableFilter = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the threshold for pulse widths to ignore when <see cref="EnableFilter"/> is set to <see langword="true"/>.
+        /// </summary>
+        public byte FilterThreshold 
+        { 
+            get => _filterThreshold; 
+            set => _filterThreshold = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the timeout threshold for the <see cref="ReceiverChannel.GetAllItems"/> call. Defaults to 1 second.
+        /// </summary>
+        public TimeSpan ReceiveTimeout 
+        { 
+            get => _receiveTimeout; 
+            set => _receiveTimeout = value;
```

Comment:
any minimum/maximum value? In that case, better to check we're in the proper range here or on the native side. And a nice description in remarks and/or exception intellisense comments is good for the developer as well

Reply:
None, according to the IDF docs. This value is used as a parameter when reading RMT items from the ringbuffer using `xRingbufferReceive` and the docs say it accepts a number of ticks.